Tests/Swift Tests/BugsnagSwiftPublicAPITests.swift

```diff
@@ -0,0 +1,368 @@
+    static func clearMetadata(section sectionName: String, key: String) {}
+    
+    func addMetadata(_ metadata: [AnyHashable : Any], section sectionName: String) {}
+    func addMetadata(_ metadata: Any?, key: String, section sectionName: String) {}
+    func getMetadata(section sectionName: String) -> NSMutableDictionary? { return NSMutableDictionary() }
+    func getMetadata(section sectionName: String, key: String) -> Any? { return nil }
+    func clearMetadata(section sectionName: String) {}
+    func clearMetadata(section sectionName: String, key: String) {}
+}
+
+class BugsnagSwiftPublicAPITests: XCTestCase {
+
+    let apiKey = "01234567890123456789012345678901"
+    let ex = NSException(name: NSExceptionName("exception"),
+                         reason: "myReason",
+                         userInfo: nil)
+    let err = NSError(domain: "dom", code: 123, userInfo: nil)
+    let sessionBlock: BugsnagOnSessionBlock = { (session) -> Bool in return false }
+    let onSendErrorBlock: BugsnagOnSendErrorBlock = { (event) -> Bool in return false }
+    let onBreadcrumbBlock: BugsnagOnBreadcrumbBlock = { (breadcrumb) -> Bool in return false }
+    
+    func testBugsnagClass() throws {
+        Bugsnag.start(withApiKey: "")
+        Bugsnag.start(withApiKey: apiKey);
+        Bugsnag.start(with: BugsnagConfiguration(apiKey))
+        
+        let _ = Bugsnag.appDidCrashLastLaunch()
+        
+        Bugsnag.notify(ex)
+        Bugsnag.notify(ex) { (event) -> Bool in return false }
+        Bugsnag.notifyError(err)
+        Bugsnag.notifyError(err) { (event) -> Bool in return false }
+        
+        Bugsnag.leaveBreadcrumb(withMessage: "msg")
+        Bugsnag.leaveBreadcrumb(forNotificationName: "notif")
+        Bugsnag.leaveBreadcrumb("msg", metadata: ["foo" : "bar"], type: .error)
+        
+        Bugsnag.startSession()
+        Bugsnag.pauseSession()
+        Bugsnag.resumeSession()
+        
+        Bugsnag.setContext("ctx")
+        let _ = Bugsnag.context()
+        
+        Bugsnag.setUser("me", withEmail: "memail@foo.com", andName: "you")
+        let _ = Bugsnag.user()
+        
+        Bugsnag.addOnSession(block: sessionBlock)
+        Bugsnag.removeOnSession(block: sessionBlock)
+        
+        Bugsnag.addOnSendError(block: onSendErrorBlock)
+        Bugsnag.removeOnSendError(block: onSendErrorBlock)
+        
+        Bugsnag.addOnBreadcrumb(block: onBreadcrumbBlock)
+        Bugsnag.removeOnBreadcrumb(block: onBreadcrumbBlock)
+    }
+    
+    func testBugsnagConfigurationClass() throws {
+        let _ = BugsnagConfiguration.loadConfig()
+        let config = BugsnagConfiguration(apiKey)
+
+        config.apiKey = apiKey
+        config.releaseStage = "stage1"
+        config.enabledReleaseStages = nil
+        config.enabledReleaseStages = ["one", "two", "three"]
+        config.redactedKeys = nil
+        config.redactedKeys = ["1", 2, 3]
+        config.redactedKeys = ["a", "a", "b"]
```

Comment:
`config.redactedKeys` should also allow `NSRegularExpression` objects in the same array